src/scalability/data-sharding.md

```diff
@@ -0,0 +1,23 @@
+# Data sharding
+
+## What is the partition
+
+For fault tolerance, TiKV replicates multiple copies of the same data on different nodes based on Raft. But for that large dataset or high query throughput, all data is managed by one Raft state machine is not efficient and flexible, so we need to break the data up into partitions, also known as `sharding`. And we call a partition here is called a `Region` in TiKV.
+
+With partitioning, we can move partitions to balance the load among all nodes of a cluster. For how to decide which partitions to store on which nodes isn't covered here. It will be discussed later in chapter related to PD.
+
+## Partitioning
+
+Different partitions can be placed on different nodes, thus a large dataset can be distributed across many disks, and the query load can be distributed across many processors. But partitioning suffers from hot spot issue that high load may end up on one partition which makes only one node busy but other nodes idle. To avoid that, we can simply assign records to nodes randomly, but when reading a particular item there is no fast way of knowing which node is on. 
```

Comment:
```suggestion
Different partitions can be placed on different nodes, thus a large dataset can be distributed across many disks, and the query load can be distributed across many processors. But partitioning suffers from the issue of *hot spots*, where high load may end up on one partition but leave the other nodes idle. To avoid that, we can simply assign records to nodes randomly, but when reading a particular item there is no fast way of knowing which node is on. 
```